src/test/scala/org/ergoplatform/mining/ErgoMinerSpec.scala

```diff
@@ -61,6 +69,79 @@ class ErgoMinerSpec extends TestKit(ActorSystem()) with FlatSpecLike with ErgoTe
       blocksGenerated should be >= height
     }
   }
+
+  ignore should "filter out double spend txs" in {
+    val tx = validErgoTransactionGen.sample.get._2
+    ErgoMiner.fixTxsConflicts(Seq(tx, tx, tx)) should have length 1
+
+    val inputs = validErgoTransactionGenTemplate(0, -1, 100, 100).sample.get._1
+    val (l, r) = inputs.splitAt(50)
+    val tx_1 = validTransactionGen(l).sample.get
+    val tx_2 = validTransactionGen(r :+ l.last).sample.get
+
+    ErgoMiner.fixTxsConflicts(Seq(tx_1, tx_2, tx)) should contain theSameElementsAs Seq(tx_1, tx)
+    ErgoMiner.fixTxsConflicts(Seq(tx_2, tx_1, tx)) should contain theSameElementsAs Seq(tx_2, tx)
+  }
+
+  it should "work correctly with 2 coinbase txs in pool" in {
+    val tmpDir = createTempDir
+
+    val defaultSettings: ErgoSettings = ErgoSettings.read(None).copy(directory = tmpDir.getAbsolutePath)
+
+    val nodeSettings = defaultSettings.nodeSettings.copy(mining = true,
+      stateType = StateType.Utxo,
+      miningDelay = defaultAwaitDuration,
+      offlineGeneration = true,
+      verifyTransactions = true)
+    val chainSettings = defaultSettings.chainSettings.copy(blockInterval = 2.seconds)
+    val ergoSettings = defaultSettings.copy(nodeSettings = nodeSettings, chainSettings = chainSettings)
+
+    val nodeViewHolderRef: ActorRef = ErgoNodeViewRef(ergoSettings, timeProvider, emission)
+    val readersHolderRef: ActorRef = ErgoReadersHolderRef(nodeViewHolderRef)
+    val minerRef: ActorRef = ErgoMinerRef(ergoSettings, nodeViewHolderRef, readersHolderRef, timeProvider, emission)
+
+    val state: UtxoState = await(
+      (nodeViewHolderRef ? GetDataFromCurrentView[ErgoHistory, UtxoState, ErgoWallet, ErgoMemPool, UtxoState]
+        (v => v.state)).mapTo[UtxoState]
+    )
+
+    val minerProp = DLogProverInput(BigIntegers
+      .fromUnsignedByteArray(ergoSettings.scorexSettings.network.nodeName.getBytes())
+    ).publicImage
+
+    val fakeProp = DLogProverInput(BigIntegers.fromUnsignedByteArray("ttttttttt".getBytes())).publicImage
+
+    val emissionBox = state.emissionBoxOpt.get
+    val tx = ErgoMiner.createCoinbase(emissionBox, state.stateContext.height, Seq.empty, fakeProp, emission)
```

Comment:
Where is the second transaction in the pool? It is assumed to be generated by the miner?